### ISSUE TITLE
请问训练代码里的 basemodel 是什么作用？

### ISSUE THREAD

**Issue by jewelcai** [YCG09/chinese_ocr#183]
请问训练代码里的 basemodel 是什么作用？
我看代码在 loadweights 之后，没有后续的操作，请问这个 basemodel 是干嘛的呢？
另外，我想在作者提供的 weights 基础之上，添加训练集，继续训练，应该怎么训练呢？

`    modelPath = './models/pretrain_model/keras.h5'
    if os.path.exists(modelPath):
        print("Loading model weights...")
        basemodel.load_weights(modelPath)
        print('done!')`

**Comment by fire717** [comment]
我觉得可能是因为共用了y_pred这个output，所以实际是一个模型的两个不同片段的变量。

我也对这个问题很好奇   期待作者答复

**Comment by xinyuegtxy** [comment]
这作者代码写的真恶心

**Comment by xinyuegtxy** [comment]
不过我发现一个问题，就是这样：

	basemodel = Model(inputs=input, outputs=y_pred)
	basemodel = multi_gpu_model(basemodel, gpus=4)

	# basemodel.summary()

	labels = Input(name='the_labels', shape=[None], dtype='float32')
	input_length = Input(name='input_length', shape=[1], dtype='int64')
	label_length = Input(name='label_length', shape=[1], dtype='int64')

	loss_out = Lambda(ctc_lambda_func, output_shape=(1,), name='ctc')([y_pred, labels, input_length, label_length])

	model = Model(inputs=[input, labels, input_length, label_length], outputs=loss_out)
	model = multi_gpu_model(model, gpus=4)

如果不这么写，那么代码就出错.呵呵。
@jewelcai @fire717 @YCG09

**Comment by fire717** [comment]
> 不过我发现一个问题，就是这样：
> 
> ```
> basemodel = Model(inputs=input, outputs=y_pred)
> basemodel = multi_gpu_model(basemodel, gpus=4)
> 
> # basemodel.summary()
> 
> labels = Input(name='the_labels', shape=[None], dtype='float32')
> input_length = Input(name='input_length', shape=[1], dtype='int64')
> label_length = Input(name='label_length', shape=[1], dtype='int64')
> 
> loss_out = Lambda(ctc_lambda_func, output_shape=(1,), name='ctc')([y_pred, labels, input_length, label_length])
> 
> model = Model(inputs=[input, labels, input_length, label_length], outputs=loss_out)
> model = multi_gpu_model(model, gpus=4)
> ```
> 如果不这么写，那么代码就出错.呵呵。
> @jewelcai @fire717 @YCG09

你的意思是说，用多GPU训练，要写成这种复杂的形式才行吗？
但是我用的时候，直接这样：
`basemodel_multi = keras.utils.multi_gpu_model(basemodel,gpus=gpu_num)`
是没问题的，训练测试都跑过的。


另外，我觉得说别人开源的代码恶心还是有点不大合适，有问题可以探讨，你觉得不好可以分享你的，或者提PR。顺便我是与作者利益无关的一个路人。

**Comment by JJoohhnnn** [comment]
请问 你可以在博主训练好的模型上继续训练吗？我加上之后acc还是从0开始，好像是网络模型不对应？

**Comment by jewelcai** [comment]
> 请问 你可以在博主训练好的模型上继续训练吗？我加上之后acc还是从0开始，好像是网络模型不对应？

我是自己生成图片从头训练的   但是也可以在我训练出来的模型基础上继续训练

**Comment by JJoohhnnn** [comment]
> > 请问 你可以在博主训练好的模型上继续训练吗？我加上之后acc还是从0开始，好像是网络模型不对应？
> 
> 我是自己生成图片从头训练的 但是也可以在我训练出来的模型基础上继续训练

请问你的数据集里的汉字是从语料库里生成的还是从字典里随机抽取的没有关联的汉字？

**Comment by jewelcai** [comment]
> > > 请问 你可以在博主训练好的模型上继续训练吗？我加上之后acc还是从0开始，好像是网络模型不对应？
> > 
> > 
> > 我是自己生成图片从头训练的 但是也可以在我训练出来的模型基础上继续训练
> 
> 请问你的数据集里的汉字是从语料库里生成的还是从字典里随机抽取的没有关联的汉字？

自己找的语料库

**Comment by JJoohhnnn** [comment]
> > > > 请问 你可以在博主训练好的模型上继续训练吗？我加上之后acc还是从0开始，好像是网络模型不对应？
> > > 
> > > 
> > > 我是自己生成图片从头训练的 但是也可以在我训练出来的模型基础上继续训练
> > 
> > 
> > 请问你的数据集里的汉字是从语料库里生成的还是从字典里随机抽取的没有关联的汉字？
> 
> 自己找的语料库

语料库可以给我发一份吗，我找不到好的，谢谢啦  1767013330@qq.com

**Comment by mayidu** [comment]
@jewelcai 您好！想问问如何在自己训练的模型的基础上重新训练呢，我加载自己训练的模型作为预训练模型，但还是从0开始训练

**Comment by YCG09** [comment]
@fire717 你说的差不多，就是通过y_pred关联起来的，basemodel是densenet的网络结构部分，因此加载权重时候用的是basemodel；model是加入了ctc loss用来训练的实例化对象，他的模型结构部分就是basemodel，把graph保存下来打开看看就明白了
